pkg/alertmanager/operator.go

```diff
@@ -920,6 +920,9 @@ receivers:
 	}
 
 	generator := newConfigGenerator(namespacedLogger, version, store)
+	if err := generator.generateGlobalConfig(ctx, baseConfig, am, amConfigs); err != nil {
```

Comment:
I think that the operator should branch earlier depending on whether `spec.globalAlertmanagerConfig` is defined or not:
* if defined, retrieve the global `AlertmanagerConfig` object and bail out if something returns an error (missing ref, invalid object).
* otherwise work with the raw configuration secret.